packages/reakit/src/Popover/__examples__/HoverPopover/index.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import { Button } from "reakit/Button";
+import {
+  usePopoverState,
+  Popover,
+  PopoverDisclosure,
+  PopoverArrow,
+  PopoverInitialState,
+} from "reakit/Popover";
+import { Separator } from "reakit/Separator";
+import { VisuallyHidden } from "reakit/VisuallyHidden";
+
+function useHoverPopoverState(initialState?: PopoverInitialState) {
+  const timeout = 300;
+  const showTimeout = React.useRef<number | null>(null);
+  const hideTimeout = React.useRef<number | null>(null);
+  const popover = usePopoverState(initialState);
+  const clearTimeouts = React.useCallback(() => {
+    if (showTimeout.current !== null) {
+      window.clearTimeout(showTimeout.current);
+    }
+    if (hideTimeout.current !== null) {
+      window.clearTimeout(hideTimeout.current);
+    }
+  }, []);
+  const show = React.useCallback(() => {
+    clearTimeouts();
+    showTimeout.current = window.setTimeout(() => {
+      popover.show();
+    }, timeout);
+  }, [clearTimeouts, popover.show]);
+  const hide = React.useCallback(() => {
+    clearTimeouts();
+    hideTimeout.current = window.setTimeout(() => {
+      popover.hide();
+    }, timeout);
+  }, [clearTimeouts, popover.hide]);
+  React.useEffect(
+    () => () => {
+      clearTimeouts();
+    },
+    [clearTimeouts]
+  );
+  return {
+    ...popover,
+    show,
+    hide,
+  };
+}
+
+type User = {
+  fullname: string;
+  username: string;
+  description: string;
+};
+
+const user: User = {
+  fullname: "John Doe",
+  username: "@JohnDoe",
+  description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+};
+
+export default function HoverPopover() {
+  const [isFollowing, setFollowing] = React.useState(false);
+  const popover = useHoverPopoverState();
+  const popoverProps = {
+    onMouseEnter: popover.show,
+    onMouseLeave: popover.hide,
+  };
+  return (
+    <>
+      <PopoverDisclosure
+        as="a"
+        href="#"
+        aria-label={`See the complete profile of ${user.username}`}
+        {...popover}
+        {...popoverProps}
+      >
+        {user.username}
+      </PopoverDisclosure>
```

Comment:
I'm not in favor of a link which is designed for navigating, I would keep a button instead?

Reply:
I think it's okay to use a button. But I wonder why it would be opening the popover on hover, and not on click? I can only think of real-world examples of this with links (like Twitter and Facebook usernames).